### DISCUSSION TITLE
Enable `versionBump` for Docker or Regex manager

### DISCUSSION THREAD

**Discussion by jan-molak** [renovatebot/renovate#24924] [Suggest an Idea]
### What would you like Renovate to be able to do?

I'd like Renovate to automatically bump the version of the Docker image for which dependencies are getting updated. This way I could avoid a manual step in the release process.

### If you have any ideas on how this should be implemented, please tell us here.

To my understanding, there's no standard, universally applicable way to specify the version of a Docker image being built. From what I've seen, people typically use an env variable, hard-code the version in the pipeline configuration file, store the version in some auto-generated file on a disk, etc.

In my case, I configure the `VERSION` in my `.gitlab-ci.yml` file, from where it's then picked up by the release process and used as tag for the Docker image.
```yml
variables:
  VERSION: 1.2.0
```
Since I have (self-hosted) Renovate configured to propose MRs to update the base image of my image, I'd like to be able to specify a regex pattern for Renovate to use when searching for the version (similarly to how [Regex manager](https://docs.renovatebot.com/configuration-options/#matchstrings) works). For example:
```json
{
  "bumpVersion": "patch",
  "bumpVersionStrategy": {
    "fileMatch": ["^.gitlab-ci.yml$"],
    "matchStrings": [
      "VERSION:\\s(?<version>.*?)"
    ]
  }
}
```
This would enable Renovate to "know" where the version is specified, without having to add support for every manager individually.

Of course, there are some assumptions here, like for example the version following semver conventions.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by Finkregh** [comment]
Perhaps consider implementing it like https://docs.renovatebot.com/modules/datasource/#repology-datasource - then you could bump the version in multiple places

From the example above:

```yaml
variables:
  # renovate: bump-on-changes
  VERSION: 1.2.0
```

Perhaps even with an optional file limiter, that you bump place X only if the software used there has changed (... perhaps that's BS...)
